common/src/test/kotlin/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketCoreImplTest.kt

```diff
@@ -0,0 +1,690 @@
+import timber.log.Timber
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class WebSocketCoreImplTest {
+    private lateinit var mockConnection: WebSocket
+    private lateinit var webSocketCore: WebSocketCoreImpl
+    private lateinit var webSocketListener: WebSocketListener
+
+    @BeforeEach
+    fun setUp() {
+        Timber.plant(ConsoleLogTree)
+        ConsoleLogTree.verbose = true
+    }
+
+    @AfterEach
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    private fun TestScope.setupServer(url: String = "https://io.ha", backgroundScope: CoroutineScope = this.backgroundScope) {
+        val mockOkHttpClient = mockk<OkHttpClient>(relaxed = true)
+        val mockServerManager = mockk<ServerManager>(relaxed = true)
+        val mockAuthenticationRepository = mockk<AuthenticationRepository>(relaxed = true)
+
+        val testServerId = 1
+        val testServer = Server(
+            id = testServerId,
+            _name = "Test Server",
+            connection = ServerConnectionInfo(
+                externalUrl = url,
+            ),
+            session = ServerSessionInfo(),
+            user = ServerUserInfo(),
+        )
+
+        mockConnection = mockk(relaxed = true)
+        every { mockServerManager.getServer(any<Int>()) } returns testServer
+        every { mockServerManager.authenticationRepository(testServerId) } returns mockAuthenticationRepository
+        coEvery { mockAuthenticationRepository.retrieveAccessToken() } returns "mock_access_token"
+        // The queue is a channel if we don't use a background scope the test hangs because the channel is never close
+        // In this configuration the channel will be close when the test will end
+        webSocketCore =
+            WebSocketCoreImpl(
+                mockOkHttpClient,
+                mockServerManager,
+                testServerId,
+                this,
+                backgroundScope,
+            )
+        webSocketListener = webSocketCore
+        every {
+            mockOkHttpClient.newWebSocket(
+                any(),
+                webSocketListener,
+            )
+        } returns mockConnection
+    }
+
+    private fun prepareAuthenticationAnswer(
+        haVersion: String = "2025.4.1",
+        successfulAuth: Boolean = true,
+    ) {
+        // Simulate that the server receive the auth message and answer that it is accepted
+        every {
+            mockConnection.send(
+                match<String> { message ->
+                    message.contains(""""type":"auth"""") &&
+                        message.contains(""""access_token":"mock_access_token"""")
+                },
+            )
+        } answers {
+            assertSame(WebSocketState.AUTHENTICATING, webSocketCore.getConnectionState())
+            webSocketListener.onMessage(
+                mockConnection,
+                """{"id":1,"type":"${if (successfulAuth) "auth_ok" else "auth_invalid"}","ha_version":"$haVersion"}""",
```

Comment:
This is wrong? The auth message response does not include an ID

![image](https://github.com/user-attachments/assets/8d2d13aa-3c58-4b0f-8945-a56c2d51f3c8)

Reply:
Yes, it was just ignored by the parser I'm removing it 👍🏻